hls-plugin-api/src/Ide/PluginUtils.hs

```diff
@@ -241,6 +243,13 @@ allLspCmdIds pid commands = concatMap go commands
   where
     go (plid, cmds) = map (mkLspCmdId pid plid . commandId) cmds
 
+
+-- | Lookup the plugin that exposes a particular command
+lookupPluginId :: CommandId -> IdePlugins a -> Maybe PluginId
+lookupPluginId cmd (IdePlugins ls) = fst <$> find go ls
+  where
+    go (_, desc) = cmd `elem` map commandId (pluginCommands desc)
```

Comment:
This looks inefficient, I hope this doesn't end up in any hot-loop.

Reply:
Good catch, thanks. I've made it so that it will only be called once per completion call.

Reply:
Why not create an inverted index `(CommandId -> PluginId)`? Persisted inside `IdePlugins`

Reply:
I've done this now.